### PR TITLE
Delete redundant immigration entry date and restructured immigration status code and bring support in line with new changes

### PR DESCRIPTION
**Pull request by avinhurry**
## Context
We deployed the last of the immigration changes work in #6657 

This is a follow up PR to remove feature flags and redundant code which we no longer have a need for. This PR also prepares for the removal of 4 columns on the `ApplicationForm` table.

## Changes proposed in this pull request
 - Remove `:restructured_immigration_status` and `:immigration_entry_date` feature flags, including all overhanging app and spec logic (including the data migrations).

- Drop `immigration_entry_date` views, controller, form and routes 
- Drop `right_to_work` views, controller and form
- Remove redundant routes, replace old references and support routes with `immigration_right_to_work`
- Rename `Support RightToWorkOrstudy` to `SupportImmigrationRightToWork`
- Fix validation error in support when no audit comment is provided
- Tell active record to remove the following columns from the cache: `immigration_status_details`, `immigration_entry_date`, `immigration_route`, `immigration_route_details`
- Rework and fix specs
-  Bring support in line with the changes: Add form, view, controller, route for immigration status and spec

## Guidance to review

- Recommend reviewing commit by commit  
- Play around with `immigration_status`, `right_to_work` and `right_to_work_details` in support.
- Anything missing?
- Everything still work?

## Link to Trello card
https://trello.com/c/qLPBzUpR/4526-delete-immigration-entry-date-and-restructured-immigration-status-feature-flags

## Things to check

- [x] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [x] This code does not rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: app/controllers/support_interface/application_forms/immigration_status_controller.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        @immigration_status_form = ImmigrationStatusForm.build_from_application(@application_form)
+      end
+
+      def update
+        @immigration_status_form = ImmigrationStatusForm.new(immigration_status_params)
+
+        if @immigration_status_form.save(@application_form)
+          redirect_to support_interface_application_form_path(@application_form)
+        else
+          redirect_to support_interface_application_form_edit_immigration_status_path(@application_form)
```
Comment (avinhurry) [inline app/controllers/support_interface/application_forms/immigration_status_controller.rb:16]: Couldn't quite find a robust solution here. If I `render :edit` the view changes to the non eu version as `@immigration_status_form.eu_nationality?` must change to false for some reason.

Any ideas?

Comment (frankieroberto) [reply app/controllers/support_interface/application_forms/immigration_status_controller.rb:16]: I'd guess that on line 11 which does `ImmigrationStatusForm.new(immigration_status_params)` it also need to do `ImmigrationStatusForm.build_from_application(@application_form)` and then somehow merge in the params?

File: spec/system/candidate_interface/entering_details/candidate_entering_personal_details_when_requiring_visa_2022_spec.rb [hunk 2]
```diff
@@ -101,17 +92,13 @@ def and_i_can_change_nationality_to_an_eu_country_with_settled_status
...
-    fill_in 'Year', with: '2001'
-    click_button t('save_and_continue')
+    expect(page).not_to have_content 'When did you enter the UK?'
 
     expect(page).to have_current_path candidate_interface_personal_details_show_path
     expect(page).to have_content "Nationality\nFrench"
     expect(page).to have_content "Do you have the right to work or study in the UK?\nYes"
     expect(page).to have_content "Immigration status\nEU settled status"
-    expect(page).to have_content "Date of entry into the UK\n3 February 2001"
+    expect(page).not_to have_content 'Date of entry into the UK'
```
Comment (frankieroberto) [inline spec/system/candidate_interface/entering_details/candidate_entering_personal_details_when_requiring_visa_2022_spec.rb:101]: Maybe this line could be dropped, as no one will have ever answered this question...?

File: spec/system/candidate_interface/entering_details/candidate_entering_personal_details_when_requiring_visa_2022_spec.rb [hunk 3]
```diff
@@ -101,17 +92,13 @@ def and_i_can_change_nationality_to_an_eu_country_with_settled_status
     choose 'EU settled status'
     click_button t('save_and_continue')
 
-    expect(page).to have_content 'When did you enter the UK?'
-    fill_in 'Day', with: '3'
-    fill_in 'Month', with: '2'
-    fill_in 'Year', with: '2001'
-    click_button t('save_and_continue')
+    expect(page).not_to have_content 'When did you enter the UK?'
```
Comment (frankieroberto) [inline spec/system/candidate_interface/entering_details/candidate_entering_personal_details_when_requiring_visa_2022_spec.rb:95]: Maybe this line could be dropped, as no one will have ever seen this question...?

File: app/views/support_interface/application_forms/immigration_status/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+<% content_for :before_content, govuk_back_link_to(support_interface_application_form_path(@application_form)) %>
+
+<div class="govuk-grid-row">
+  <div class="govuk-grid-column-two-thirds">
+    <%= form_with model: @immigration_status_form, url: support_interface_application_form_edit_immigration_status_path, method: :patch do |f| %>
+      <%= f.govuk_error_summary %>
+      <% if @immigration_status_form.eu_nationality? %>
+        <%= f.govuk_radio_buttons_fieldset(
+        :immigration_status,
+        legend: { text: 'Edit applicant residency status', size: 'l' },
```
Comment (frankieroberto) [inline app/views/support_interface/application_forms/immigration_status/edit.html.erb:11]: We should update this wording to match the question.

```suggestion
        legend: { text: 'Edit applicant immigration status', size: 'l' },
```

File: app/views/support_interface/application_forms/immigration_status/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,45 @@
+<% content_for :browser_title, title_with_error_prefix('Edit applicant residency status', @immigration_status_form.errors.any?) %>
```
Comment (frankieroberto) [inline app/views/support_interface/application_forms/immigration_status/edit.html.erb:1]:
```suggestion
<% content_for :browser_title, title_with_error_prefix('Edit applicant immigration status', @immigration_status_form.errors.any?) %>
```

File: app/views/support_interface/application_forms/immigration_status/edit.html.erb [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        :immigration_status,
+        legend: { text: 'Edit applicant residency status', size: 'l' },
+      ) do %>
+        <%= f.govuk_radio_button :immigration_status, 'eu_settled', label: { text: 'EU settled status' }, link_errors: true %>
+        <%= f.govuk_radio_button :immigration_status, 'eu_pre_settled', label: { text: 'EU pre-settled status' } %>
+        <%= f.govuk_radio_button :immigration_status, 'other', label: { text: 'Other' } do %>
+          <%= f.govuk_text_area(
+                :right_to_work_or_study_details,
+                label: { text: 'Immigration status', class: 'govuk-label govuk-label--s' },
+                hint: { text: 'For example, “I have permanent residence”' },
```
Comment (frankieroberto) [inline app/views/support_interface/application_forms/immigration_status/edit.html.erb:19]: Updated with a more accurate example - and dropping the "I have" prefix.
```suggestion
                hint: { text: 'For example, “Indefinite leave to remain”' },
```

File: app/views/support_interface/application_forms/immigration_status/edit.html.erb [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+                threshold: 90,
+              ) %>
+        <% end %>
+      <% end %>
+      <% else %>
+        <%= f.hidden_field :immigration_status, value: 'other' %>
+        <%= f.govuk_text_area(
+              :right_to_work_or_study_details,
+              label: { text: 'Immigration status', size: 'l' },
+              hint: { text: 'For example, “I have permanent residence”' },
```
Comment (frankieroberto) [inline app/views/support_interface/application_forms/immigration_status/edit.html.erb:30]:
```suggestion
              hint: { text: 'For example, “Indefinite leave to remain”' },
```

**Review by frankieroberto — APPROVED** [review]
Tested on review app, and everything still worked as expected... 🙌 

One small unrelated thing, I forgot to mention before that we should update the hint text for the immigration status free text field to "For example, 'Indefinite leave to remain'" rather than "For example, 'I have permanent residency'" as we don’t think that "permanent residency" is technically an immigration status, and candidates don't need to put "I have" at the start.

File: app/views/support_interface/application_forms/immigration_status/edit.html.erb (hunk 5 above)
Comment (avinhurry) [reply app/views/support_interface/application_forms/immigration_status/edit.html.erb:1]: Thanks @frankieroberto 

Should we also change the row text in support? Not sure of the context why it uses residency instead.

image

**Comment by avinhurry** [comment]
> Tested on review app, and everything still worked as expected... 🙌
> 
> One small unrelated thing, I forgot to mention before that we should update the hint text for the immigration status free text field to "For example, 'Indefinite leave to remain'" rather than "For example, 'I have permanent residency'" as we don’t think that "permanent residency" is technically an immigration status, and candidates don't need to put "I have" at the start.

Thanks @frankieroberto! All those changes have been made

**Review by stevehook — APPROVED** [review]
Looks good to me. I think you already fixed the thing I commented on!

File: app/controllers/support_interface/application_forms/immigration_status_controller.rb (hunk 1 above)
Comment (avinhurry) [reply app/controllers/support_interface/application_forms/immigration_status_controller.rb]: Yep you do @stevehook. That was a temp fix until I figured it out. All fixed now.

Thanks both! ♻️

File: app/views/support_interface/application_forms/immigration_status/edit.html.erb (hunk 5 above)
Comment (frankieroberto) [reply app/views/support_interface/application_forms/immigration_status/edit.html.erb:1]: @avinhurry yeah, I spotted that too. I've fixed in https://github.com/DFE-Digital/apply-for-teacher-training/pull/6694 but feel free to fix it in this PR too. 👍

Comment (avinhurry) [reply app/views/support_interface/application_forms/immigration_status/edit.html.erb:1]: I've fixed! You will need to rebase 😓. I'll review your PR

**Pull request merged**